lib/webidl2.js

```diff
@@ -154,12 +199,17 @@
       if (typeof value === "undefined" || tokens[0].value === value) {
         last_token = tokens.shift();
         line += count(last_token.trivia, "\n");
-        if (type === ID && last_token.value.startsWith('_'))
-          last_token.value = last_token.value.substring(1);
         return last_token;
       }
     }
 
+    function unescape(identifier) {
+      if (identifier.startsWith('_')) {
```

Comment:
Nit: maybe just?:
```JS
return identifier.startsWith('_') ? identifier.slice(1) : identifier; 
```

then there is no reassignment. But no problem to keep it as is.